### PR TITLE
fix(query): get next send sequence for transfer packets

### PR DESCRIPTION
**Pull request by dzmitry-lahoda**
## Description

closes: #4698

### Commit Message / Changelog Entry

```bash
feat(packet): get next send sequence for transfer packets
```
Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [x] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [x] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [x] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [x] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

**Comment by dzmitry-lahoda** [comment]
what else should I do?

**Review by crodriguezvega — COMMENTED** [review]
Thanks, @dzmitry-lahoda. It would be good to fix [this test](https://github.com/cosmos/ibc-go/actions/runs/6224990849/job/16898568050?pr=4706). You can do something like it is done [in this other test for ordered channels](https://github.com/cosmos/ibc-go/blob/fec13d938993eaaa62af369999d6fbc2f79835ba/modules/core/04-channel/keeper/grpc_query_test.go#L1535-L1550), where the next sequence send is set and then the test check that the query returns the expected value.

File: modules/core/04-channel/keeper/grpc_query_test.go [hunk 1]
```diff
@@ -1620,12 +1620,14 @@ func (suite *KeeperTestSuite) TestQueryNextSequenceSend() {
...
 		},
 		{
-			"basic success on unordered channel returns zero",
+			"basic success on unordered channel returns the set send sequence",
 			func() {
 				path := ibctesting.NewPath(suite.chainA, suite.chainB)
 				suite.coordinator.Setup(path)
 
-				expSeq = 0
+				expSeq = 42
```
Comment (crodriguezvega) [inline modules/core/04-channel/keeper/grpc_query_test.go:1628]: The answer to the ultimate question of life, the universe, and everything! ❤️

**Review by crodriguezvega — APPROVED** [review]
Looks good to me, @dzmitry-lahoda!

File: modules/core/04-channel/keeper/grpc_query_test.go [hunk 2]
```diff
@@ -1620,12 +1620,14 @@ func (suite *KeeperTestSuite) TestQueryNextSequenceSend() {
...
+				seq := uint64(expSeq)
```
Comment (crodriguezvega) [inline modules/core/04-channel/keeper/grpc_query_test.go:1629]:
```suggestion
				seq := uint64(42)
```

File: modules/core/04-channel/keeper/grpc_query.go [hunk 3]
```diff
@@ -555,25 +555,21 @@ func (k Keeper) NextSequenceSend(c context.Context, req *types.QueryNextSequence
...
 	ctx := sdk.UnwrapSDKContext(c)
 
-	channel, found := k.GetChannel(ctx, req.PortId, req.ChannelId)
+	_, found := k.GetChannel(ctx, req.PortId, req.ChannelId)
 	if !found {
 		return nil, status.Error(
 			codes.NotFound,
 			errorsmod.Wrapf(types.ErrChannelNotFound, "port-id: %s, channel-id %s", req.PortId, req.ChannelId).Error(),
 		)
 	}
```
Comment (damiannolan) [inline modules/core/04-channel/keeper/grpc_query.go:564]: I think this block of code can be removed. There is no need to check the channel exists here if the channel itself is unused.

**Review by damiannolan — COMMENTED** [review]
Thank you @dzmitry-lahoda! I think we can remove the check on the channel lookup

File: modules/core/04-channel/keeper/grpc_query.go (hunk 3 above)
Comment (dzmitry-lahoda) [reply modules/core/04-channel/keeper/grpc_query.go:564]: i retained old code for conaistency of errors and ease of debug.

Comment (dzmitry-lahoda) [reply modules/core/04-channel/keeper/grpc_query.go:564]: i guess getchannel method at some point could be redesigned.

Comment (damiannolan) [reply modules/core/04-channel/keeper/grpc_query.go:564]: Thank you, appreciate the attention and care for retaining the previous behaviour wrt errors etc. 

I would still recommend to remove the channel lookup as its unnecessary. If the channel does not exist then likewise, the `nextSequenceSend` will not be set and an error will be returned to the client.

**Review by damiannolan — APPROVED** [review]
Thank you @dzmitry-lahoda! LGTM 🚀

**Comment by colin-axner** [comment]
It looks like the pr will need to be brought up to date with main, if you allow maintainers write access (usually a button you can click on the side), we can manage this for you

**Comment by josietyleung** [comment]
@colin-axner @damiannolan will this be included in v8 of ibc go? thanks https://github.com/cosmos/ibc-go/blob/main/docs/docs/01-ibc/09-roadmap.md

**Comment by damiannolan** [comment]
> @colin-axner @damiannolan will this be included in v8 of ibc go? thanks https://github.com/cosmos/ibc-go/blob/main/docs/docs/01-ibc/09-roadmap.md

Hey @josietyleung, yes it will be included in the v8 release 👍

**Pull request merged**